lib/azure/core/http/retry_policy.rb

```diff
@@ -22,7 +22,9 @@ module Http
       # specific policy when HTTP layer errors occur
       class RetryPolicy < HttpFilter
 
-        def initialize(&block)
+        def initialize(options = {}, &block)
+          @retry_exceptions = options[:retry_exceptions] || [::StandardError]
```

Comment:
When `:retry_exceptions` are specified, customers may only focus on the errors other than the standard errors, so they may not have `::StandardError` included. But `::StandardError` is we want to handle anyway. 

So I'd suggest:
1. To merge `::StandardError` with `options[:retry_exceptions]`
2. Ensure each item in `options[:retry_exceptions]` is an `Exception`.